app/components/buttons/EyeFilterMenuWithSlider/index.js

```diff
@@ -0,0 +1,152 @@
+import { EyeFilterMenu } from "buttons";
+import { FormattedMessage as T } from "react-intl";
+import "style/EyeFilterMenu.less";
+import "style/MiscComponents.less";
+import noUiSlider from "nouislider";
+import {  NumericInput } from "inputs";
+
+@autobind
+class EyeFilterMenuWithSlider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      minAmount: 0,
+      maxAmount: 100,
+      min: 0,
+      max: 100,
+      sliderShower: true,
+      expandedSliderInfo: false,
+    };
+  }
+
+  getSliderWhenOpeningMenu() {
+    setTimeout(() => {
+      const range = document.getElementById("min-max-slider");
+
+      const { minAmount, maxAmount, min, max } = this.state;
+      const toolTipFormatter = {
+        to: (value) => {
+          return value.toFixed(2);
+        },
+      };
+
+      noUiSlider.create(range, {
+        start: [ minAmount, maxAmount ],
+        range: {
+          "min": [ min ],
+          "max": [ max ]
+        },
+        connect: true,
+        tooltips: [ true, toolTipFormatter ],
+      });
+
+      range.noUiSlider.on("end", (values, handle) => {
+        const value = values[handle];
+        if (handle) {
+          this.props.onChangeSlider(value, "max");
+          this.setState({ maxAmount: value });
+        } else {
+          this.setState({ minAmount: value });
+          this.props.onChangeSlider(value, "min");
+        }
+      });
+
+      range.noUiSlider.on("update", (values, handle) => {
+        const value = values[handle];
+        if (handle) {
+          this.setState({ maxAmount: value });
+        } else {
+          this.setState({ minAmount: value });
+        }
+      });
+
+      this.setState({ rangeSlider: range });
+    }, 25);
+  }
+
+  getSliderWhenOpenedMenu() {
+    const { currencyDisplay } = this.props;
+
+    const { sliderShower, expandedSliderInfo,
+      min, max, maxAmount, minAmount } = this.state;
+
+    return (
+      <div className="history-slider-wrapper">
+        <div className="history-amount-range-label"><T id="history.amount.range" m="Amount Range" /></div>
+        <div id="min-max-slider" className="min-max-slider"></div>
```

Comment:
Here, you'd use:

```suggestion
        <div id="min-max-slider" className="min-max-slider" ref={r => this.createSliderRangeInElement(r)}></div>
```

So when the ref is established, the `noUiSlider` lib will be invoked to create the slider in this particular ref.